x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndex.java

```diff
@@ -37,6 +37,14 @@ public static String jobResultsIndexPrefix() {
         return AnomalyDetectorsIndexFields.RESULTS_INDEX_PREFIX;
     }
 
+    /**
+     * The name of the alias pointing to the indices where the jobs' results are stored
+     * @return The read alias
+     */
+    public static String jobResultsIndexName() {
+        return AnomalyDetectorsIndexFields.RESULTS_INDEX_PREFIX + AnomalyDetectorsIndexFields.RESULTS_INDEX_DEFAULT;
```

Comment:
This always returns `.ml-anomalies-shared`.  So I guess the intention is to store the datafeed stats for all jobs in the shared results index.  But for jobs with custom indices this is the first time we've ever stored documents relating to a job in an index other than the custom index specified for the job.  I think this will cause problems in the future, for example we will forget this subtlety when deciding which indices to search for documents relating to a job if we implement job export/import.

So I think this PR should be refactored to store datafeed timings in the current write index for the associated job.